### PR TITLE
feat(typeahead): add 'is-toggled' option

### PR DESCRIPTION
**Pull request by chenyuzhcy**
This is related to issue #4760

### CODE REVIEW

**Comment by RobJacobs** [comment]
@chenyuzhcy I'm assuming you are adding a feature to indicate whether the typeahead dropdown is open or not?  In all the other directives that offer this feature we have been using 'is-open' so for consistency sake, could we name this the same?

File: src/typeahead/test/typeahead.spec.js [hunk 1]
```diff
@@ -588,6 +588,78 @@ describe('typeahead tests', function() {
...
+          expect($scope.isToggled).toBeTruthy();
+          changeInputValueTo(element, 'not match');
+          expect($scope.isToggled).toBeFalsy();
+      });
+
+      it('should bind is-toggled indicator as false when a match is clicked', function () {
+          expect($scope.isToggled).toBeFalsy();
+          changeInputValueTo(element, 'b');
+          expect($scope.isToggled).toBeTruthy();
+          var match = $(findMatches(element)[1]).find('a')[0];
```
Comment (wesleycho) [inline src/typeahead/test/typeahead.spec.js]: Better to do `var match = findMatches(element).find('a').eq(0)`

File: src/typeahead/test/typeahead.spec.js [hunk 2]
```diff
@@ -588,6 +588,78 @@ describe('typeahead tests', function() {
...
+
+          $(match).click();
```
Comment (wesleycho) [inline src/typeahead/test/typeahead.spec.js]: Given the above change, should be able to do just `match.click()`

**Comment by wesleycho** [comment]
This LGTM, and I'm fine merging while making the changes as merging - any objections here?

**Comment by chenyuzhcy** [comment]
@RobJacobs I had the same idea at first but then I looked all the 'is-open' in the other directives, it all does toggle to the dropdown, but this is only an indicator of open or not (it does not manually open the dropdown even we set it to true) so I chose the name that's mentioned in the issue. But if you still insist 'is-open' is a better name I can change it very quickly.

**Comment by wesleycho** [comment]
I think `is-open` is a better name - I think the semantic is off in the other directives, but too late for those.

**Pull request closed without merging**